src/ui/SubredditPage.tsx

```diff
@@ -16,30 +16,37 @@ export default class SubredditPage extends Page implements common.SubredditView
     <ViewModeToggleAction page={this}/>
   );
 
-  private _items: RedditPost[] = [];
   private _mode: ViewMode;
   private loading: boolean;
-  @getById private collectionView: CollectionView;
+  @property private _items: List<RedditPost> = new List();
+  @getById private listView: ListView<RedditPost>;
 
   constructor(properties?: Properties<SubredditPage>) {
     super(properties);
     this.append(
-      <CollectionView id='collectionView'
+      <ListView
+          id='listView'
           stretch
           background='#f5f5f5'
-          cellType={() => this._mode}
-          cellHeight={(_index, type) => isList(type) ? 96 : 160}
-          createCell={(type: string) => this.createCell(type)}
-          updateCell={(view, index) => (view as Cell).item = this._items[index].data}
-          onLastVisibleIndexChanged={this.handleLastVisibleIndexChanged}/>
+          bind-items='_items'
+          onLastVisibleIndexChanged={this.handleLastVisibleIndexChanged}
+          onSelect={({item}) => this.onItemSelected.trigger({item: item as RedditPost})}>
+        <RedditListCell
+            height={96}
+            selectable
+            itemCheck={() => this.mode === ViewMode.List}/>
+        <RedditGalleryCell
+            height={160}
+            selectable
+            itemCheck={() => this.mode !== ViewMode.List}/>
```

Comment:
Ah, excuse my own indecisiveness, but what would be even better is `this.mode === ViewMode.Gallery`.